File: leetcode/1003.cpp
class Solution {
public:
    bool isValid(string str) {
        vector<char> s;
        for (int i=0; i<str.length();++i){
            s.push_back(str[i]);
            if (
                s.size() >=3 && 
                s[s.size()-3]=='a' &&
                s[s.size()-2]=='b' &&
                s[s.size()-1]=='c'
            ) {
                for (int j = 0; j<3;++j) s.pop_back();
            }
        }
        return s.empty();
    }
};

File: leetcode/1003.java
class Solution {
    public boolean isValid(String S) {
        ArrayList<Character> a = new ArrayList<>();
        for (int i=0;i<S.length();++i){
            a.add(S.charAt(i));
            if (a.size() >= 3 &&
                    a.get(a.size() - 3).equals('a') &&
                    a.get(a.size() - 2).equals('b') &&
                    a.get(a.size() - 1).equals('c')
            ) {
                a.remove(a.size()-1);
                a.remove(a.size()-1);
                a.remove(a.size()-1);
            }
        }
        return a.isEmpty();
    }
}
